test/torch/mpc/falcon/test_falcon_helper.py

```diff
@@ -1,24 +1,121 @@
+import pytest
+
 import torch
 from syft.frameworks.torch.mpc.falcon.falcon_helper import FalconHelper
 
+import itertools
+
+
+TEST_VALS = [(x, y, x ^ y) for x, y in itertools.product(torch.LongTensor([0, 1]), repeat=2)]
+
 
-def test_public_xor(workers):
+@pytest.mark.parametrize("x, y, x_xor_y", TEST_VALS)
+def test_public_xor(x, y, x_xor_y, workers):
     bob, alice, james = (workers["bob"], workers["alice"], workers["james"])
-    x = torch.tensor([0]).share(bob, alice, james, protocol="falcon", field=2)
-    y = torch.tensor([1])
-    a = torch.tensor([1]).share(bob, alice, james, protocol="falcon", field=2)
-    b = torch.tensor([0])
-    assert (FalconHelper.xor(x, y).reconstruct() == torch.tensor(1)).all()
-    assert (FalconHelper.xor(x, b).reconstruct() == torch.tensor(0)).all()
-    assert (FalconHelper.xor(a, y).reconstruct() == torch.tensor(0)).all()
-    assert (FalconHelper.xor(a, b).reconstruct() == torch.tensor(1)).all()
+    x = x.share(bob, alice, james, protocol="falcon", field=2)
+    assert (FalconHelper.xor(x, y).reconstruct() == x_xor_y).all()
 
 
-def test_private_xor(workers):
+@pytest.mark.parametrize("x, y, x_xor_y", TEST_VALS)
+def test_private_xor(x, y, x_xor_y, workers):
     bob, alice, james = (workers["bob"], workers["alice"], workers["james"])
-    x = torch.tensor([0]).share(bob, alice, james, protocol="falcon", field=2)
+    x = x.share(bob, alice, james, protocol="falcon", field=2)
+    y = y.share(bob, alice, james, protocol="falcon", field=2)
+    assert (FalconHelper.xor(x, y).reconstruct() == x_xor_y).all()
+
+
+def test_xor_invalid_ring_size(workers):
+    bob, alice, james = (workers["bob"], workers["alice"], workers["james"])
+
+    x = torch.tensor([0]).share(bob, alice, james, protocol="falcon", field=3)
     y = torch.tensor([1]).share(bob, alice, james, protocol="falcon", field=2)
-    assert (FalconHelper.xor(x, y).reconstruct() == torch.tensor(1)).all()
-    assert (FalconHelper.xor(x, x).reconstruct() == torch.tensor(0)).all()
-    assert (FalconHelper.xor(y, y).reconstruct() == torch.tensor(0)).all()
-    assert (FalconHelper.xor(y, x).reconstruct() == torch.tensor(1)).all()
+
+    with pytest.raises(ValueError) as e:
```

Comment:
last request: let's move the assertion inside the body of the code, the tests should only test the interface and helper functions, we can use assertions in code during construction it would make it easier to find and remove them before moving it to production
for example, put (assert x.ring_size == 2) inside the xor function instead of if statements and remove the corresponding test case 
also better manage whitespace, related expressions should be in a continuous chunk

Reply:
It is not ok to have asserts in the "real" code because they can be removed if you run with the ```-O``` or ```-OO``` flag.

Reply:
yes that's the point, you should remove the asserts before pushing to production, asserts makes it easier to do that if you want your checks not to be moved to production to improve performance